src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/RenameHandler.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    [Shared]
+    [ExportLspMethod(Methods.TextDocumentRenameName)]
+    internal class RenameHandler : IRequestHandler<RenameParams, WorkspaceEdit>
```

Comment:
Shouldn't this also be a PrepareRename handler so we don't get the rename dialog for every piece of text?

Reply:
Unfortunately, VS LSP platform doesn't support PrepareRename requests yet. I am going to follow up if they have any plans for doing it soon.